### DISCUSSION TITLE
Prime Provider dosnt work on prime components that come from external lib

### DISCUSSION THREAD

**Discussion by ArtisanDuWeb1** [primefaces/community#1465] [PrimeReact]
Hello, I'm currently attempting to create a bundle using PrimeReact as an external dependency.

My issue is that when using the bundle, the PrimeReactProvider imported into the apps does not provide custom styles to the components coming from my bundle, even though they are within the provider. However, it does provide styles to components imported directly from Prime in the app.

If I export the Provider from the bundle, the bundle's components are styled, but not the app's ones.

Could you please advise if there is a way to resolve this?

**Comment by orelus** [comment]
Hi !

I got a same problem, can we have a workaround ?

**Reply by ArtisanDuWeb1** [reply]
Hello!

I'm still struggling to get it right.

The solution I've opted for (which is a bit messy) is to wrap each
prime component that I export in a provider. This approach brings
along several issues, such as managing the z-index, which I had to
address with custom CSS.

As for the ongoing issue, I suspect it might be related to the
instance of the prime context. It seems like the context instances in
the bundle and the project using it might not be the same.

What are your thoughts?
